chalicelib/es_connection.py

```diff
@@ -207,7 +208,26 @@ def get_main_page_checks(self, primary=True):
         }
         search = Search(using=self.es, index=self.index)
         search.update_from_dict(doc)
-        return self.search(search)
+        raw_result = self.search(search)
+        if checks is not None:
+            # figure out which checks we didn't find, add a placeholder check so
+            # that check is still rendered on the UI
+            found = {}
+            for check_name in checks:
+                found[check_name] = False
+            for result in raw_result:
+                found[result['name']] = True
+            for check_name, has_record in found.items():
```

Comment:
This can be simplified to something like:
```
found_checks = [res['name'] for res in raw_result]
for check_name in checks:
    if check_name not in found_checks:
        < create and add placedholder >
```

Reply:
Is the n^2 runtime not a big deal here?

Reply:
I guess you could use a `set` to store `found_checks` in the code I wrote so the lookup is faster. But it's up to you, I'm cool with the old code if you want to keep it

Reply:
Going with set because its cleaner